POL-172 Update AWS policies for credential Service 2

KTebrewal opened the pull request:
### Description

Update listed AWS policies to support the credential service.  The changes include-
-Replace auth declaration with the credential declaration.
-Remove long_description field and use new info hash for Version, providers and service.
-Update tenant to single.
-Update readme to replace credential instructions with instructions to use credential service.
-Update CWF definitions as needed. 

### Issues Resolved

Changed the authentication to credential services.

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### Running template Link

AWS Savings Plan Recommendations - https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e144c8e372f7a0001f736df

![image](https://user-images.githubusercontent.com/55429424/71887518-4279f180-3164-11ea-8105-8a1c1404589c.png)

AWS Subnet Name Tag Sync-
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e26aa1706101000014d560e

AWS Unencrypted ELB Listeners (ALB/NLB)-
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e26abe006101000014d5610

AWS Unencrypted ELB Listeners (CLB)-
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e26ae10e4ca220001b69e3d

AWS Unencrypted RDS Instances - https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e203d8f32401200016e02f3
![image](https://user-images.githubusercontent.com/55429424/72519238-f8d48980-387c-11ea-8581-5f954169992a.png)

AWS Unencrypted Volumes - https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e26af244ec7b300014e66c0

![image](https://user-images.githubusercontent.com/55429424/72785802-09f11200-3c52-11ea-8b94-c1236c88358d.png)

AWS Unused ECS Clusters-
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e26a1eb4ec7b300014e66a7
![image](https://user-images.githubusercontent.com/55429424/72791567-8b9a6d00-3c5d-11ea-862a-edd30beb077f.png)

AWS VPC Name Tag Sync-
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e26c338e4ca220001b69e51

AWS Unencrypted S3 Buckets-
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e26d01a4ec7b300014e66ed
![image](https://user-images.githubusercontent.com/55429424/72796995-4f1f3f00-3c66-11ea-8af5-c00c493f3700.png)

cdwilhelm commented on compliance/aws/ecs_unused/AWS_Unused_ECS_Clusters.pt:
```diff
@@ -5,6 +5,13 @@ short_description "Report and remediate any ECS clusters that are not currently
 long_description "Version: 1.2"
 category "Compliance"
 severity "low"
+tenancy "single"
```
tenancy can be removed

cdwilhelm commented on operational/aws/subnet_name_sync/README.md:
```diff
@@ -1,39 +1,37 @@
-## AWS Subnet Name Tag Sync Policy Template
+# AWS Subnet Name Tag Sync Policy Template
```
Remove Policy Template from Name

cdwilhelm commented on operational/aws/subnet_name_sync/README.md:
```diff
@@ -1,39 +1,37 @@
 This policy has the following input parameter required when launching the policy.
 
-- *Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
```
Missing leading *

cdwilhelm commented on operational/aws/subnet_name_sync/aws_subnet_name_sync.pt:
```diff
@@ -2,9 +2,16 @@ name "AWS Subnet Name Tag Sync"
+long_description ""
 severity "medium"
 category "Operational"
+tenancy "single"
```
remove tenancy

cdwilhelm commented on operational/aws/subnet_name_sync/aws_subnet_name_sync.pt:
```diff
@@ -33,11 +40,11 @@ end
+credentials "auth_aws" do
+  schemes "aws"
+  label "AWS"
+  description "Select the AWS IAM Credential from the list"
```
remove IAM.  
The credentail can use the AWS IAM or STS credentials.

cdwilhelm commented on operational/aws/vpc_name_sync/README.md:
```diff
@@ -1,40 +1,37 @@
-## AWS VPC Name Tag Sync Policy Template
+# AWS VPC Name Tag Sync Policy Template
```
please remove policy template

cdwilhelm commented on security/aws/clb_unencrypted/README.md:
```diff
@@ -1,27 +1,32 @@
+## Input Parameters
+
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Ignore tags* - CLB with any of these tags will be ignored
```
Missing the leading *

cdwilhelm commented on security/aws/clb_unencrypted/aws_clb_encryption.pt:
```diff
@@ -1,10 +1,17 @@
+long_description ""
 category "Security"
 severity "medium"
+tenancy "single"
```
remove tenancy

cdwilhelm commented on security/aws/clb_unencrypted/aws_clb_encryption.pt:
```diff
@@ -35,11 +42,11 @@ end
+credentials "auth_aws" do
+  schemes "aws"
+  label "AWS"
+  description "Select the AWS IAM Credential from the list"
```
Remove IAM

cdwilhelm commented on security/aws/ebs_unencrypted_volumes/AWS_Unencrypted_Volumes.pt:
```diff
@@ -5,6 +5,14 @@ short_description "Report any Elastic Block Store (EBS) volumes in AWS that are
 long_description "Version: 1.1"
 category "Security"
 severity "low"
+tenancy "single"
```
remove tenancy

cdwilhelm commented on security/aws/ebs_unencrypted_volumes/AWS_Unencrypted_Volumes.pt:
```diff
@@ -35,11 +43,11 @@ end
+credentials "auth_aws" do
+  schemes "aws"
+  label "AWS"
+  description "Select the AWS IAM Credential from the list"
```
remove IAM

cdwilhelm commented on security/aws/ebs_unencrypted_volumes/README.md:
```diff
@@ -1,26 +1,31 @@
+## Input Parameters
+
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Ignore tags* - EBS volumes with any of these tags will be ignored
```
Missing the leading *

cdwilhelm commented on security/aws/elb_unencrypted/README.md:
```diff
@@ -1,27 +1,32 @@
+## Input Parameters
+
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Ignore tags* - ALB/NLB with any of these tags will be ignored
```
Missing the leading *

cdwilhelm commented on security/aws/elb_unencrypted/aws_elb_encryption.pt:
```diff
@@ -2,9 +2,16 @@ name "AWS Unencrypted ELB Listeners (ALB/NLB)"
+long_description ""
 category "Security"
 severity "medium"
+tenancy "single"
```
remove tenancy

cdwilhelm commented on security/aws/rds_unencrypted/AWS_Unencrypted_RDS_Instances.pt:
```diff
@@ -2,9 +2,16 @@ name "AWS Unencrypted RDS Instances"
+long_description ""
 category "Security"
 severity "medium"
+tenancy "single"
```
Remove tenancy

cdwilhelm commented on security/aws/rds_unencrypted/README.md:
```diff
@@ -1,70 +1,73 @@
+## Input Parameters
+
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Ignore tags* - RDS instances with any of these tags will be ignored
```
Missing leading *

cdwilhelm commented on security/aws/unencrypted_s3_buckets/AWS_Unencrypted_S3_Buckets.pt:
```diff
@@ -1,10 +1,17 @@
+long_description ""
 category "Security"
 severity "low"
+tenancy "single"
```
remove tenancy

cdwilhelm commented on security/aws/unencrypted_s3_buckets/AWS_Unencrypted_S3_Buckets.pt:
```diff
@@ -35,11 +42,11 @@ end
+credentials "auth_aws" do
+  schemes "aws"
+  label "AWS"
+  description "Select the AWS IAM Credential from the list"
```
remove IAM

KTebrewal commented:
@cdwilhelm - I will address these comments once i am done with an integration priority task. Anyways this PR was still under progress. I still had to make changes. I missed out leaving a comment. Will request a review once it is updated.

KTebrewal commented:
@cdwilhelm - While this PR is still in progress, i have two doubts here.
1) I am getting errors in Danger file like - Policy Template path should be lowercase. security/aws/unencrypted_s3_buckets/AWS_Unencrypted_S3_Buckets.pt
To change the Policy name cases, i deleted the files from my repository and renamed in lower case, but it still does not change the cases in policy name. How can we make the changes here?
2) You had asked to remove IAM for only a few policies. It should be for only those or i should apply that for all the policies?

cdwilhelm commented:
@KTebrewal   When referencing IAM for the credentials IAM should be removed in all cases as the policies should be able to support AWS IAM user and AWS STS (cross-account).  

Regarding the filename, I'm guessing you are windows and the default file naming is case insensitive.  You probably need to disable case insensitive settings on your system.  You can try to google how to change file name case on windows.

cdwilhelm commented:
@KTebrewal   we found the AWS policies credential schemes requires "aws","aws_sts"  Can you please update the policies with the "aws","aws_sts" schemes.  thanks

KTebrewal commented:
@cdwilhelm - I have addressed all the comments. There is one issue with the credentials -
i have added schema for AWS IAM and AWS STS credentials, but when i test it in UI, i only see the credentials for AWS IAM.
![image](https://user-images.githubusercontent.com/55429424/72800162-ae804d80-3c6c-11ea-8b87-bd44b6301628.png)
